models/classes/theme/ThemeService.php

```diff
@@ -102,10 +102,14 @@ public function removeThemeById($themeId)
             return false;
         }
 
-        $themes = $this->getOption(static::OPTION_AVAILABLE);
-        unset($themes[$themeId]);
+        $availableThemes = $this->getOption(static::OPTION_AVAILABLE);
+        foreach ($this->getAllThemes() as $key => $theme) {
+            if ($theme->getId() === $themeId) {
+                unset($availableThemes[$key]);
+            }
+        }
```

Comment:
As I saw the $key is always the themeId, so you can reduce your code to:

```suggestion
        $availableThemes = $this->getOption(static::OPTION_AVAILABLE);
        
        unset($availableThemes[$themeId]);
```

The unset will not throw any error in case the index does not exist.

Reply:
That's almost exactly the code before the changes ;)

The PR description suggests the theme ID gets postfixed with an index or something.

Reply:
The issue is exactly that. The `$key` is not the `$themeId`. A theme can be added multiple times and it is postfixed with an id: https://github.com/oat-sa/tao-core/blob/master/models/classes/theme/ThemeServiceAbstract.php#L133
Keys are `bosa_selor_default_theme0` and `bosa_selor_default_theme1` etc.
So, the idea here is to request the id from the theme class and if it is the same as the id what should be removed, remove it.

Reply:
🤦  True, I did not analyse it right. I've approved the PR :)